src/main/java/com/google/cloud/tools/StageGenericJavaAction.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools;
+
+import com.google.common.io.Files;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Stages a Java JAR/WAR Managed VMs application to be deployed.
+ */
+public class StageGenericJavaAction extends Action {
+
+  private File appYaml;
+  private File dockerfile;
+  private File artifact;
+  private File stagingDir;
+
+  public StageGenericJavaAction(File appYaml, File dockerfile, File artifact,
+      File stagingDir) {
+    this.appYaml = appYaml;
```

Comment:
With the Maven plugin, I was actually never passing null. I always passed the assumed default location of app.yaml, which might or might not exist. However, I understand that other users might just want to pass in a null (e.g., in case they have no assumed default location of app.yaml).

I'll add a check for null too. Let me know if that does it for you.